### PR TITLE
Lead opening proposal whitelist limit (#2491)

### PR DESCRIPTION
**Pull request by chrlschwb**
Issue #2491.
When creating a WG lead opening, a whitelist limit of 20 was set, but the limit was not activated.
Created a new variable to implement this.
Error message "Maximum whitelist size of 20 members is reached." added.

### CODE REVIEW

File: packages/ui/src/proposals/modals/AddNewProposal/components/TriggerAndDiscussionStep.tsx [hunk 1]
```diff
@@ -24,7 +26,12 @@ export const TriggerAndDiscussionStep = () => {
   ])
 
   const addMemberToWhitelist = (member: Member) => {
-    setValue('triggerAndDiscussion.discussionWhitelist', [...discussionWhitelist, member], { shouldValidate: true })
+    if (count < 20){
+      setValue('triggerAndDiscussion.discussionWhitelist', [...discussionWhitelist, member], { shouldValidate: true })
+      setCount(count+1)
+    }else{
+      setSelected(true)
+    }
```
Comment (traumschule) [inline packages/ui/src/proposals/modals/AddNewProposal/components/TriggerAndDiscussionStep.tsx:34]: Please extend existing validation on the select field like [here](https://github.com/Joystream/pioneer/blob/dev/packages/ui/src/proposals/modals/AddNewProposal/components/TriggerAndDiscussionStep.tsx#L75), see also [machine.ts](https://github.com/Joystream/pioneer/blob/dev/packages/ui/src/proposals/modals/AddNewProposal/machine.ts)

**Review by traumschule — CHANGES_REQUESTED** [review]
This looks ok on itself but we need to follow the implemented logic. Marking this as draft. Would add to description: Fixes #2491 (follow-up to #3855)
cc @WRadoslaw

**Review by WRadoslaw — CHANGES_REQUESTED** [review]
Hey! Thanks for your contribution! 

About the issue, I would prefer to keep validation strictly in the form vendor. Please check out `packages/ui/src/proposals/modals/AddNewProposal/helpers.ts` line `173` I suppose. The discussion whitelist is already validated there, so it's only about adding an additional method on `yup`. This may come in handy -> https://github.com/jquense/yup#array

File: packages/ui/src/proposals/modals/AddNewProposal/components/TriggerAndDiscussionStep.tsx [hunk 2]
```diff
@@ -1,4 +1,4 @@
-import React from 'react'
```
Comment (WRadoslaw) [inline packages/ui/src/proposals/modals/AddNewProposal/components/TriggerAndDiscussionStep.tsx:1]: This file do not require any changes

File: packages/ui/src/proposals/modals/AddNewProposal/helpers.ts [hunk 3]
```diff
@@ -171,7 +171,7 @@ export const schemaFactory = (api?: ProxyApi) => {
       }),
       isDiscussionClosed: Yup.boolean(),
       discussionWhitelist: Yup.array().when('isDiscussionClosed', {
-        is: true,
+        is: Yup.array().length() < 20,
```
Comment (WRadoslaw) [inline packages/ui/src/proposals/modals/AddNewProposal/helpers.ts:174]:
```suggestion
        is: true,
```

File: packages/ui/src/proposals/modals/AddNewProposal/helpers.ts [hunk 4]
```diff
@@ -171,7 +171,7 @@ export const schemaFactory = (api?: ProxyApi) => {
...
         then: Yup.array().required('Field is required'),
```
Comment (WRadoslaw) [inline packages/ui/src/proposals/modals/AddNewProposal/helpers.ts:175]:
```suggestion
        then: Yup.array().max(20, 'Maximum whitelist size of ${max} members is reached.').required('Field is required'),
```

**Review by WRadoslaw — CHANGES_REQUESTED** [review]

File: packages/ui/src/proposals/modals/AddNewProposal/helpers.ts (hunk 4 above)
Comment (chrlschwb) [reply packages/ui/src/proposals/modals/AddNewProposal/helpers.ts]: changed accordingly

**Comment by traumschule** [comment]
There's too much wrong here, please recreate after testing locally. @gyroflaw please help out with pioneer setup.
- deleted yarn.lock
- added node as dependency
- removed react from imports
- not using prettier
```
15:01:06.689 | \| src/utils.ts(11,22): error TS2769: No overload matches this call.
15:01:06.689 | \| Overload 1 of 2, '(o: {}): string[]', gave the following error.
15:01:06.689 | \| Argument of type 'T' is not assignable to parameter of type '{}'.
15:01:06.689 | \| Overload 2 of 2, '(o: object): string[]', gave the following error.
15:01:06.689 | \| Argument of type 'T' is not assignable to parameter of type 'object'.
15:01:06.717 | \| error Command failed with exit code 2.
```

**Pull request closed without merging**